src/node.cpp

```diff
@@ -142,4 +148,35 @@ std::vector<session_key_t> Node::getReadySessions() const {
     return std::move(readySessions);
 }
 
+Status Node::demultiplyOutputs(SessionResults& nodeSessionOutputs) {
+    auto& [metadata, blobMap] = nodeSessionOutputs.begin()->second;
```

Comment:
beside generatingNewSubsessions all below should happen per each output. One node can have mutliple outputs eg ["a", "b"], so we should demultiply each output blob separately.